src/providers/scripts/results.ts

```diff
@@ -57,6 +61,73 @@
+                    }
+                    // Reset and show the timer content
+                    timerElement.textContent = 'Time elapsed:';
+                    timerElement.style.display = 'block';
+                    clearInterval(timer);
+                    timer = setInterval(() => {
+                        elapsedTime = Date.now() - startTime;
+                        let displayTime;
+
+                        // If less than 1000 ms, show full milliseconds
+                        if (elapsedTime < 1000) {
+                            displayTime = `${elapsedTime}ms`;
+                        } else {
+                            // After 1000 ms, display seconds with one decimal place
+                            const seconds = (elapsedTime / 1000).toFixed(1);
+                            const totalSeconds = Math.floor(elapsedTime / 1000);
+                            const minutes = Math.floor(totalSeconds / 60);
+                            const remainingSeconds = totalSeconds % 60;
+
+                            if (minutes > 0) {
+                                // Display minutes with one decimal place for the seconds
+                                displayTime = `${minutes}m ${remainingSeconds.toString().padStart(2, '0')}s`;
+                            } else {
+                                // Display only seconds with one decimal place
+                                displayTime = `${seconds}s`;
+                            }
+                        }
+
+                        if (timerElement) {
+                            timerElement.textContent = `Time elapsed: ${displayTime}`;
+                        }
+
+                        // Update the timer every 100ms before 1 second, and every 500ms after 1 second
+                        if (elapsedTime >= 1000 && timer) {
+                            clearInterval(timer);
+                            timer = setInterval(() => {
+                                elapsedTime = Date.now() - startTime;
+                                const seconds = (elapsedTime / 1000).toFixed(1);
+                                const totalSeconds = Math.floor(elapsedTime / 1000);
+                                const minutes = Math.floor(totalSeconds / 60);
+                                const remainingSeconds = totalSeconds % 60;
+
+                                if (minutes > 0) {
+                                    displayTime = `${minutes}m ${remainingSeconds.toString().padStart(2, '0')}s`;
+                                } else {
+                                    displayTime = `${seconds}s`;
+                                }
+
+                                if (timerElement) {
+                                    timerElement.textContent = `Time elapsed: ${displayTime}`;
+                                }
+                            }, 500);
+                        }
+                    }, elapsedTime < 1000 ? 100 : 500);
```

Comment:
The same rate as in the beginning looks great for the seconds too! And maybe you can also simplify the logic by having both with the same rate.

```suggestion
                    }, 103);
```

I changed from `100` to `103`, to avoid repetitive numbers during the increases.